modules/product/src/main/java/com/opengamma/strata/product/swap/SwapIndex.java

```diff
@@ -0,0 +1,354 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product.swap;
+
+import java.io.Serializable;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanBuilder;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.strata.collect.named.Named;
+import com.opengamma.strata.product.swap.type.FixedIborSwapTemplate;
+
+/**
+ * Class describing a swap index.
+ */
+@BeanDefinition(builderScope = "private")
+public final class SwapIndex
```

Comment:
I think `SwapIndex` should follow the same pattern as `IborIndex`, with an interface, implementation, use of extended enums and a CSV loader.